plugins/lua/zone.lua

```diff
@@ -1216,10 +1216,188 @@ function RetireLocationOverlay:confirm_retire()
         curry(retire, location))
 end
 
+-- -------------------
+-- AnimalActionsWidget
+--
+
+local YES = 'Yes'
+local NO = 'No'
+
+-- Make sure an animal unit of your civ is selected
+local function check_valid_unit()
+    local unit = dfhack.gui.getSelectedUnit(true)
+    return unit
+        and dfhack.units.isFortControlled(unit)
+        and dfhack.units.isAlive(unit)
+        and dfhack.units.isAnimal(unit)
+end
+
+local function is_geldable()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    -- If the unit is not male or already gelded do nothing
+    return dfhack.units.isGeldable(unit)
+end
+
+local function is_not_pet()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    return not dfhack.units.isPet(unit)
+end
+
+local function is_avail_adoption()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    -- Make sure not already a pet and not a cat as they adopt owner
+    return not dfhack.units.isPet(unit) and unit.race ~= 171
```

Comment:
This won't be true for all worlds. Race IDs will change according to the mods you have loaded.

Is there a race trait we can check against? If not, you might have to compare against the string id (`CAT`)

Reply:
I found a method that caravan uses and slightly modified it and seems to work.